docs/how_to_dev_frontend_en.md

```diff
@@ -0,0 +1,105 @@
+npm reads `package.json` to build different package according to the packaging script.
+
+Here is a snippet of the package.json
+```
+{
+  "name": "VisualDL",
+  "version": "1.0.0",
+  "description": "Visualization toolkit for deep learning",
+  "scripts": {
+    "release": "cross-env NODE_ENV=production node ./tool/build.js",
+    "build": "cross-env NODE_ENV=dev node ./tool/build.js",
+    "dev": "cross-env NODE_ENV=dev node tool/dev-server.js",
+    "lint": "./node_modules/fecs/bin/fecs --rule"
+  },
+  "engines": {
+    "node": ">= 6.4.0"
+  },
+  "dependencies": {
+    "axios": "^0.16.1",
+    "csshint": "^0.3.3"
+    ...
+    ...
+    ...
+  }
+}
+```
+
+
+## Development Flow
+The VisualDL is an open source project. Everyone is welcome to contribute to the project. To get start the process, please refer to [Contribute Code](contribute_code_en.md) to set up the environment.
+
+1. Create Development Server
+
+    Once the setup up is complete. You may create a local development server to test new features or bug fixes.
+
+    To create a development server, please run the following commands.
+
+    ```bash
+    # Navigate to the web-app folder
+    cd VisualDL/frontend
+    # Use npm to run the dev script
+    npm run dev
+    ```
+
+    You should see the follow lines
+    ```bash
+    > VisualDL@1.0.0 dev /Users/YOUR_WORK_PLACE/VisualDL/frontend
+    > cross-env NODE_ENV=dev node tool/dev-server.js
+
+    2018-01-30 18:53:36 [INFO] enable autoresponse middleware...
+    2018-01-30 18:53:37 [INFO] export global mock helper variable mock
+    > Starting dev server...
+    webpack: wait until bundle finished:
+    ...
+    ...
+    ...
+    webpack: Compiled successfully.
+    > Listening at http://localhost:8999
+    ```
+
+    This means a development server is ready and is accessible at `http://localhost:8999`. Open a browser and navigate to `http://localhost:8999` to access the VisualDL in development mode.
+    The development server will listen to any change made to the San or Javascript and update the changes in real time. We recommend doing web-app development under this mode to save time and resources.
+
+1. Prepare for Production
+
+    Once the implementation is complete. We need to run a final test by packaging all the Javascript files and minify them. Please run the following command.
+
+    ```bash
+    # Package all the needed info and update the manifest.js, vendor.js and index.js
+    npm run build
+    ```
+
+    The webpack will now append all the Javascript files and minify them. This process is to test if the files are properly packaged.
+    If there is no error, You may now check-in the modified files.
+
+    **Note:** There is no need to check-in the generated files.
+    During the production process, the Travis-CI will generate new packaged files and overwrite the previously generated files. Avoid checking in the generated files can keep the repo commit history clean.
```

Comment:
Travis-CI is an assumption; you don't know what CI tool other people are using. I would generally avoid reference to bundling and deployment processes, people have their own preferences.

Reply:
I will remove the Travis-CI detail.